rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -158,13 +158,13 @@ object Reduce {
         filteredExprs.size
       ).scanLeft(0)(_ + _)
 
-      def split(rand: Blake2b512Random, idx: Int): Blake2b512Random =
+      def split(rand: Blake2b512Random, startIdx: Int, idx: Int): Blake2b512Random =
         if (starts(6) == 1)
           rand
         else if (starts(6) > 256)
-          rand.splitShort((starts(idx) + idx).toShort)
+          rand.splitShort((starts(startIdx) + idx).toShort)
         else
-          rand.splitByte((starts(idx) + idx).toByte)
+          rand.splitByte((starts(startIdx) + idx).toByte)
```

Comment:
turns out that change wasn't correct :D Again, Ideally you'd just fixup! + rebase

Reply:
for some reason I got some rebase conflicts so it was easier to just add a new commit